src/server/services/procedures/geolocation/geolocation.js

```diff
@@ -50,23 +52,81 @@ async function reverseGeocode(lat, lon, query) {
     });
 }
 
-/**
- * Geolocates the address and returns the coordinates
- * @param {String} address target address
- * @returns {Object} structured data representing the location of the address
- */
-GeoLocationRPC.geolocate = async function (address) {
+async function rawGeolocate(address) {
     return await cache.wrap(address, async () => {
         logger.trace('Geocoding (not cached)', address);
         try {
             const res = await geocoder.geocode(address);
-            return { latitude: res[0].latitude, longitude: res[0].longitude };
+            if (res.length === 0 || !res[0]) throw Error('no results');
+            return res[0];
         } catch (e) {
             const message = `Geocoding (${address}) error: ${e.message}`;
             logger.warn(message);
             throw Error(`Failed to find location for ${address}`);
         }
     });
+}
+
+/**
+ * Geolocates the address and returns the coordinates
+ * @param {String} address target address
+ * @returns {Object} structured data representing the location of the address
+ */
+GeoLocationRPC.geolocate = async function (address) {
+    const raw = await rawGeolocate(address);
+    return { latitude: raw.latitude, longitude: raw.longitude };
+};
+
+/**
+ * Get information about the timezone of the provided address or location.
+ * You can provide either a list of two values representing the latitude and longitude location, or a string address to look up.
+ *
+ * @param {Union<Tuple<Latitude,Longitude>,String>} address target location - either a list representing [latitude, longitude], or an address string
+ * @returns {Object} information about the target's timezone
+ */
+GeoLocationRPC.timezone = async function (address) {
```

Comment:
Do we want to use a union type here or would it be better to have different RPCs for each possible signature? I am wondering bc latitude and longitude values have been treated as separate arguments prior to this. Will this slight change in convention be problematic for students? (I don't know but am wondering)

Reply:
Actually, even just using `String` would have the same effect because a string lookup of form `lat,long` (like stringified arrays) through the underlying geolocation lib is supported, so the union just avoids having to recompute the lat/long. But maybe it'd be better to just separate like you mentioned - maybe `timezoneAt(lat,long)` and `timezoneLookup(address)`? Not sure what to call them. I wish we had overloading, but with blocks it just wouldn't work...